docs/developer-docs/ai/inference.mdx

```diff
@@ -0,0 +1,63 @@
+---
+keywords: [intermediate, concept, AI, ai, deAI, deai]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Decentralized AI Inference
+
+<MarkdownChipRow labels={["Intermediate", "Concept", "DeAI" ]} />
+
+It's possible for canister smart contracts to run inference in a number of ways, depending on the decentralization and performance requirements.
+
+## 1. Inference on-chain
+
+Currently, ICP supports on-chain inference of small models using AI libraries such as [Sonos Tract](https://github.com/sonos/tract) that compile to WebAssembly.
+Check out the [image classification example](/docs/current/developer-docs/ai/ai-on-chain) to learn how it works.
+The long-term [vision of DeAI on ICP](https://internetcomputer.org/roadmap#Decentralized%20AI-start) is to support on-chain GPU compute to enable both training and inference of larger models.
+
+### Examples
+
+- [GPT2](https://github.com/modclub-app/rust-connect-py-ai-to-ic/tree/main/internet_computer/examples/gpt2): An example of GPT2 running on-chain. `Rust`
+- [ELNA AI](https://github.com/elna-ai): A fully on-chain AI agent platform and marketplace. Supports both on-chain and off-chain LLMs. [Try it here](https://dapp.elna.ai/).
+- [Tensorflow on ICP](https://github.com/carlosarturoceron/decentAI): An Azle example that uses a pre-trained model for making predictions. `TypeScript`
+- [ICGPT](https://github.com/icppWorld/icgpt): A React frontend that uses a C/C++ backend running an LLM fully on-chain. [Try it here](https://icgpt.icpp.world/). `TypeScript` `C++`
```

Comment:
```suggestion
- [ICGPT](https://github.com/icppWorld/icgpt): A React frontend that uses a C/C++ backend running an LLM fully on-chain. [Try it here](https://icgpt.icpp.world/).
```

Reply:
These tags are intentional, as it makes it easier to search for examples in specific languages.

Reply:
This isn't a format that we use across the docs to tag languages, and in most instances the tags feel redundant since the language is most often included in the description of the project. Also, I'm not sure that adding these has any meaningful impact on the search results or search functionality. 

Alternatively, we can add UI tags for these languages that does align with how we format things across the docs.